### DISCUSSION TITLE
系统无障碍列表显示 GKD 无法运行/此服务出现故障 的解决方案

### DISCUSSION THREAD

**Discussion by lisonge** [gkd-kit/gkd#814] [Announcements]
如果您在之前已经正常开启过无障碍的授权并且 GKD 正常工作过一段时间

然后突然某个时刻 GKD 不执行点击了，并且系统的无障碍列表里的 GKD 项显示类似如下的信息

```
无法运行，点按即可查看相关信息
```

这种情况有两种原因

- GKD 后台发生崩溃
- 正常运行，但是你的系统某些原因把 GKD 鲨了

解决方案是去上面的列表关闭 GKD 的无障碍授权后重新打开即可

如果你设置了 `无感保活`，简单下拉通知栏也即可自动重启

---

如果您在之前已经打开了 **设置-保存日志** ，那么 GKD 已经将崩溃记录到日志里，否则不会保存就不用往下看了

您可以将日志分享到此 issue 下，只需要到 `主页-设置-分享日志-分享文件/生成链接` 将文件或者链接发送到此 issue 下即可

**Comment by qyph** [comment]
https://github.com/gkd-kit/inspect/files/14734784/file.zip

**Comment by lisonge** [comment]
> https://github.com/gkd-kit/inspect/files/14734784/file.zip

你发送的日志里只有APP刚刚启动的输出，没有其他的输出，你是不是将 `保存日志` 关闭了？

**Comment by qyph** [comment]
等我遇到该bug再提交。

**Comment by moonbai** [comment]
https://github.com/gkd-kit/inspect/files/15401548/file.zip

**Comment by moonbai** [comment]
> https://github.com/gkd-kit/inspect/files/15401548/file.zip

https://github.com/gkd-kit/inspect/files/15478384/file.zip
今天又遇到一次

**Comment by adproqwq** [comment]
[log-1717330602327.zip](https://github.com/user-attachments/files/15525285/log-1717330602327.zip)

**Comment by moonbai** [comment]
更新1.8.0 beta2之后再次遇到无障碍自动关闭
日志:https://f.gkd.li/15935369

**Comment by moonbai** [comment]
![Screenshot_2024-07-03-16-33-15-876_com.fankes.apperrorstracking.jpg](https://github.com/gkd-kit/gkd/assets/34854875/b3d20f48-b910-4cf4-8256-8befc07ad6ca)



https://f.gkd.li/16080823

**Reply by moonbai** [reply]
> 请问这个可以查看异常日志的是什么软件？

好像是一个XP模块，叫什么名字不太记得了，好久不用了

**Comment by MrLee980** [comment]
https://f.gkd.li/16108967
最新的beta3和beta4一直无障碍故障，不到一分钟就会故障一次。手机vivo xfold3p 最新系统安卓14

**Comment by lisonge** [comment]
> https://f.gkd.li/16108967 最新的beta3和beta4一直无障碍故障，不到一分钟就会故障一次。手机vivo xfold3p 最新系统安卓14

https://github.com/gkd-kit/gkd/issues/650

**Comment by liqiao7** [comment]
https://f.gkd.li/16137464

自从更新安卓14后，几乎每天都会故障一次。

**Comment by Lyiaung** [comment]
橘子系统4.0 几个小时就终止一次
https://f.gkd.li/16191852

**Comment by MrLee980** [comment]
把规则里面把微博关了就好了&nbsp;


lovervipli
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年7月12日(星期五) 下午5:45
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [gkd-kit/gkd] 系统无障碍列表显示 `GKD 无法运行` 的解决方案 (Issue #570)














 
橘子系统4.0 几个小时就终止一次
 https://f.gkd.li/16191852

**Comment by antonchen** [comment]
[log-1725777646254.zip](https://github.com/user-attachments/files/16921106/log-1725777646254.zip)

已经完全打不开了

**Comment by swordnoworld** [comment]
[log-1731563309159.zip](https://github.com/user-attachments/files/17743779/log-1731563309159.zip)
无障碍无法启动，点击没反应

**Comment by AncestralSaltedFish** [comment]
https://f.gkd.li/18089017
荣耀手机更新到magic9过后，时灵时不灵的，只有重新开关添加到下拉菜单快捷栏的gkd快捷键才能重新使用，具体能用多久没测试过

**Reply by lisonge** [reply]
日志里没看到崩溃错误，应该是被系统杀后台了

**Reply by AncestralSaltedFish** [reply]
后台锁，电池优化，自启动全部设置了，还有下载了无障碍管理器进行锁定，不知道什么原因还是会被杀掉后台。

**Comment by eatumom** [comment]


**Reply by lisonge** [reply]
日志中里面没有任何错误信息，应该是被手机杀后台了

**Comment by gyj2008** [comment]
![Screenshot_2025-01-22-08-02-14-070_li.songe.gkd.jpg](https://github.com/user-attachments/assets/ad5afb2d-fd99-42fe-915e-eb2bdb3ba084)

![log-1737504357480.zip](https://github.com/user-attachments/files/18497854/log-1737504357480.zip)

1.9.3版本，每天都会出现这样情况，卸载重新安装也不行

**Comment by YOUQY979** [comment]
[log-1739763107883.zip](https://github.com/user-attachments/files/18820205/log-1739763107883.zip)
有时候能正常用几天，有时候一个小时就失效了，比如2月17号的日志，不知道是什么原因导致的。
无root，锁后台、省电、自启动都设置过了，后来改用adb高级授权还是一样，需要在无障碍关闭再重新打开才恢复。
现在下拉通知栏的按钮也没法用，会显示“关闭无障碍失败”，看不懂日志，求助！

**Comment by MakiseKurisu** [comment]
华为畅享20 pro 5G，之前都正常，系统也是停在harmony 3.0一直没有更新，突然就没法用了
[log-1742632809299.zip](https://github.com/user-attachments/files/19402092/log-1742632809299.zip)
提示访问受限，前往授权普通高级shizuku都配置过，也没用

**Reply by MakiseKurisu** [reply]
昨天已经重装过了，本来准备去obtainium降级的，结果提示当前1.9.3有更新安装，重装后结果行了...不知道怎么回事

**Comment by ajmz** [comment]
新升级到LineageOS 22.2 ，无障碍功能：此服务出现故障，以前22.1的时候是正常的。
[log-1744496599972.zip](https://github.com/user-attachments/files/19724749/log-1744496599972.zip)

**Comment by thomasmolover** [comment]
https://f.gkd.li/19850258

即使用 root 权限仍然不能自动保活，重新进入无障碍列表手动关闭再打开有时也并不能重启，需要在应用管理强制关闭彻底以后，再手动打开应用自己就能打开保活了。是不是有什么机制限制了自己调用无障碍服务。
需要彻底强制关闭，说明可能还有运行，可是不能自己打开无障碍保活，这就很尴尬。

**Comment by eatumom** [comment]
https://f.gkd.li/21478607
最新的oppo 一加 真我 无障碍权限变了
之前都运行的好好的 后台 自启动 电池优化 无障碍权限 全部给了
现在只要是打开手机后台点下清理后台 gkd的无障碍权限就会发生故障需要手动重新开启
同时用一木记账自动记账软件 更新了一个版本也好了 针对oppo系列的系统
<img width="918" height="1758" alt="一木记账" src="https://github.com/user-attachments/assets/cdc0084f-31f3-461c-bfa4-104e7506957f" />

**Reply by superorange** [reply]
同问题 解决了吗？

[a comment by an automated account is not shown]

**Comment by BH2WFR** [comment]
coloros, 几乎每天都会掉无障碍，省电优化全都开了

**Reply by DenisZheng** [reply]
同coloros, 几乎每天都会无障碍故障，要重新关掉重开

**Reply by BH2WFR** [reply]
> 同coloros, 几乎每天都会无障碍故障，要重新关掉重开

我找到解决方案了，要在设置里面找「**关联启动**」，然后把 GKD 的**关联启动**（注意这里指的不是「自启动」）打开，同时「**自启动**」中也要打开 GKD 的**自启动**

**Reply by Kanon1982** [reply]
> > 同coloros, 几乎每天都会无障碍故障，要重新关掉重开
> 
> 我找到解决方案了，要在设置里面找「**关联启动**」，然后把 GKD 的**关联启动**（注意这里指的不是「自启动」）打开，同时「**自启动**」中也要打开 GKD 的**自启动**

我用的color os 15，按照大佬说的，成功了。。但是，一旦，在后台退出gkd会出问题。。所以，现在后台，给gkd这个软件**把锁打上**，然后，打开 **gkd**--**设置**--**后台隐藏**。。。这样一来，color os 15的话，再在后台关闭应用的时候，就不会关闭gkd了
